File: ALittle/Module/ALittleIDE/Other/GameLibrary/CEngine/Script/CEngine/UISystem/Base/TextArea.lua

module("ALittle", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


local __tostring = tostring
assert(ALittle.DisplayObject, " extends class:ALittle.DisplayObject is nil")
TextArea = Lua.Class(ALittle.DisplayObject, "ALittle.TextArea")

function TextArea:Ctor(ctrl_sys)
	___rawset(self, "_text", "")
	___rawset(self, "_bold", false)
	___rawset(self, "_italic", false)
	___rawset(self, "_underline", false)
	___rawset(self, "_deleteline", false)
	___rawset(self, "_outline", false)
	___rawset(self, "_flip", 0)
	___rawset(self, "_halign_type", UIEnumTypes.HALIGN_LEFT)
	___rawset(self, "_valign_type", UIEnumTypes.VALIGN_TOP)
	___rawset(self, "_show", __CPPAPITextArea())
	A_LoadTextureManager:RegisterRedrawControl(self)
end

function TextArea:Redraw()
	self._show:NeedDraw()
end

function TextArea.__setter:font_path(value)
	self._font_path = value
	if self._font_path == nil or self._font_size == nil then
		return
	end
	self._ctrl_sys:SetFont(self, self._font_path, self._font_size)
end

function TextArea.__setter:font_size(value)
	self._font_size = value
	if self._font_path == nil or self._font_size == nil then
		return
	end
	self._ctrl_sys:SetFont(self, self._font_path, self._font_size)
end

function TextArea.__getter:font_path()
	return self._font_path
end

function TextArea.__getter:font_size()
	return self._font_size
end

function TextArea.__setter:text(value)
	if value == nil then
		return
	end
	value = __tostring(value)
	if self._text == value then
		return
	end
	self._text = value
	self._show:SetText(value)
end

function TextArea.__getter:text()
	return self._text
end

function TextArea.__setter:bold(value)
	if self._bold == value then
		return
	end
	self._bold = value
	self._show:SetBold(value)
end

function TextArea.__getter:bold()
	return self._bold
end

function TextArea.__setter:italic(value)
	if self._italic == value then
		return
	end
	self._italic = value
	self._show:SetItalic(value)
end

function TextArea.__getter:italic()
	return self._italic
end

function TextArea.__setter:underline(value)
	if self._underline == value then
		return
	end
	self._underline = value
	self._show:SetUnderline(value)
end

function TextArea.__getter:underline()
	return self._underline
end

function TextArea.__setter:deleteline(value)
	if self._deleteline == value then
		return
	end
	self._deleteline = value
	self._show:SetDeleteline(value)
end

function TextArea.__getter:deleteline()
	return self._deleteline
end

function TextArea.__setter:outline(value)
	if self._outline == value then
		return
	end
	self._outline = value
	self._show:SetOutline(value)
end

function TextArea.__getter:outline()
	return self._outline
end

function TextArea.__setter:halign(value)
	if self._halign_type == value then
		return
	end
	self._halign_type = value
	self._show:SetHAlign(value)
end

function TextArea.__getter:halign()
	return self._halign_type
end

function TextArea.__setter:valign(value)
	if self._valign_type == value then
		return
	end
	self._valign_type = value
	self._show:SetVAlign(value)
end

function TextArea.__getter:valign()
	return self._valign_type
end

function TextArea.__getter:real_height()
	return self._show:GetRealHeight()
end

function TextArea.__getter:flip()
	return self._flip
end

function TextArea.__setter:flip(value)
	self._flip = value
	self._show:SetFlip(value)
end

File: ALittleIDE/Module/ALittleIDE/Other/GameLibrary/CEngine/JSScript/CEngine/UISystem/Base/TextArea.js
{
if (typeof ALittle === "undefined") window.ALittle = {};


let __tostring = ALittle.String_ToString;
if (ALittle.DisplayObject === undefined) throw new Error(" extends class:ALittle.DisplayObject is undefined");
ALittle.TextArea = JavaScript.Class(ALittle.DisplayObject, {
	Ctor : function(ctrl_sys) {
		this._text = "";
		this._bold = false;
		this._italic = false;
		this._underline = false;
		this._deleteline = false;
		this._flip = 0;
		this._halign_type = 0;
		this._valign_type = 0;
		this._show = ALittle.NewObject(JavaScript.JTextArea);
	},
	Redraw : function() {
		this._show.NeedDraw();
	},
	set font_path(value) {
		this._font_path = value;
		if (this._font_path === undefined || this._font_size === undefined) {
			return;
		}
		this._ctrl_sys.SetFont(this, this._font_path, this._font_size);
	},
	set font_size(value) {
		this._font_size = value;
		if (this._font_path === undefined || this._font_size === undefined) {
			return;
		}
		this._ctrl_sys.SetFont(this, this._font_path, this._font_size);
	},
	get font_path() {
		return this._font_path;
	},
	get font_size() {
		return this._font_size;
	},
	set text(value) {
		if (value === undefined) {
			return;
		}
		value = __tostring(value);
		if (this._text === value) {
			return;
		}
		this._text = value;
		this._show.SetText(value);
	},
	get text() {
		return this._text;
	},
	set bold(value) {
		if (this._bold === value) {
			return;
		}
		this._bold = value;
		this._show.SetBold(value);
	},
	get bold() {
		return this._bold;
	},
	set italic(value) {
		if (this._italic === value) {
			return;
		}
		this._italic = value;
		this._show.SetItalic(value);
	},
	get italic() {
		return this._italic;
	},
	set underline(value) {
		if (this._underline === value) {
			return;
		}
		this._underline = value;
		this._show.SetUnderline(value);
	},
	get underline() {
		return this._underline;
	},
	set deleteline(value) {
		if (this._deleteline === value) {
			return;
		}
		this._deleteline = value;
		this._show.SetDeleteline(value);
	},
	get deleteline() {
		return this._deleteline;
	},
	set halign(value) {
		if (this._halign_type === value) {
			return;
		}
		this._halign_type = value;
		this._show.SetHAlign(value);
	},
	get halign() {
		return this._halign_type;
	},
	set valign(value) {
		if (this._valign_type === value) {
			return;
		}
		this._valign_type = value;
		this._show.SetVAlign(value);
	},
	get valign() {
		return this._valign_type;
	},
	get real_height() {
		return this._show.GetRealHeight();
	},
	get flip() {
		return this._flip;
	},
	set flip(value) {
		this._flip = value;
		this._show.SetFlip(value);
	},
}, "ALittle.TextArea");

}